### ISSUE TITLE
[amazon-thor] compat-wireless ath6kl doesn't work, usb probe times out

### ISSUE THREAD

**Issue by MoreRobustThanYou** [postmarketOS/pmbootstrap#932]
dmesg: https://pastebin.com/j89tSLY5

after doing ```sudo modprobe wlan``` (since IIRC compat-wireless compiles everything into a single wlan.ko), ath6kl tries to load, but fails horribly ```[43.537500] <6>ath6kl: can't wait for usb probe done```.
Firmware seems to load successfully (no errors), but that's it.

**Comment by ollieparanoid** [comment]
I've found a random other [`dmesg` log](https://pastebin.com/RjnS1Pn9) from another Android device with the same wifi chip as I've searched for the error message. The surprising part is, that it has the same message, but that doesn't seem to be an error. If I read it right, it just continues using the `wifi` driver after that.

So... have you checked if you get a wlan device after modprobing, and if you can use it?

I hope this helps 👍

**Comment by MoreRobustThanYou** [comment]
@ollieparanoid nope, and if I sit through the ~5 mins of ```sudo modprobe wlan```, I get...

```modprobe: ERROR: could not insert 'wlan': I/O error
```

**Comment by ollieparanoid** [comment]
This Linux wireless wiki has a lot of information about the `ath6kl` driver:
https://wireless.wiki.kernel.org/en/users/Drivers/ath6kl
https://wireless.wiki.kernel.org/en/users/Drivers/ath6kl/architecture

and they even list an [IRC channel](https://wireless.wiki.kernel.org/en/users/Drivers/ath6kl#irc). Since they are the experts on that matter, maybe they can help you out. Please post if you get any further or need more ideas.

**Comment by MoreRobustThanYou** [comment]
@ollieparanoid the IRC channel has 1 person on it.

**Comment by ollieparanoid** [comment]
I know I've asked you for this information already. But could you add the following to this topic? That's the most systematic approach I can currently think of.

Known working OS:
* (a1) output of `dmesg` 
* (a2) output of [`dumpsys.sh`](https://wiki.postmarketos.org/wiki/Tuning_sysfs) without being connected to wlan
* (a3) output of [`dumpsys.sh`](https://wiki.postmarketos.org/wiki/Tuning_sysfs) with being connected to wlan

postmarketOS:
* (b1) outpud of `dmesg` (already posted above, no need to do it again)
* (b2) output of [`dumpsys.sh`](https://wiki.postmarketos.org/wiki/Tuning_sysfs)

When you have that information, you should be able to extract the missing information in theory.
* diff (a2) and (a3)
* diff (a2) and (b2)
* remove timestamps of (a1) and (b1) and diff the two

**Comment by MoreRobustThanYou** [comment]
Lineage dmesg: https://pastebin.com/MkkGyFb1
Lineage dumpsys w/out wlan: https://gist.github.com/MoreRobustThanYou/b7a37187e6507876286d666aee41cabe
Lineage dumpsys w/ wlan: https://gist.github.com/MoreRobustThanYou/389d185af0bc3bf762523cb310ee1c88

pmOS sysfs: https://gist.github.com/MoreRobustThanYou/cea308e23821945aecce03a7b56f5a76

(for the sysfs dumps, they were too big for pastebin, so i used gist!)

**Comment by ollieparanoid** [comment]
As you've sent me in IRC, the diff of (a2) and (a3): https://pastebin.com/qWCKRf4J

**Comment by steamp0rt** [comment]
@ollieparanoid Would it be possible to make a script that can parse the diff and auto set the values?

**Comment by ollieparanoid** [comment]
In theory, yes. But quite some time has passed since you have generated the diff and you are using another kernel nowadays. If you would like to look into Wi-Fi on the amazon-thor some more, how about we start with a new issue with the current dmesg output, and try to figure it out from there?

**Comment by steamp0rt** [comment]
Before that, @ollieparanoid, question? How SHOULD I be building compat wireless and installing the modules? Before, I was doing ```make -C "$srcdir"/build M=$cwdir ARCH="$_carch" CC="${CC:-gcc}" CFLAGS_MODULE="-fno-pic" \
		KBUILD_BUILD_VERSION="$((pkgrel + 1))-Alpine" CONFIG_NO_ERROR_ON_MISMATCH=y \
		modules``` to build and ```make -C "$srcdir"/build M=$cwdir ARCH="$_carch" CC="${CC:-gcc}" \
		KBUILD_BUILD_VERSION="$((pkgrel + 1))-Alpine" CONFIG_NO_ERROR_ON_MISMATCH=y \
		INSTALL_MOD_PATH="$pkgdir" INSTALL_MOD_STRIP=1 modules_install
``` to install.

**Comment by ollieparanoid** [comment]
I'm not sure what's the properway to compile this kernel module (you could read up on how the other devices do it though).

However, after reading up on the [ath6kl](https://wireless.wiki.kernel.org/en/users/Drivers/ath6kl) wiki page again, I've noticed:

> This driver is upstream as part of the Linux 2.6.37 release under the staging area. From Linux 3.2 onwards it was promoted from staging to a proper wireless driver.
>
> It is also available for older kernels through the stable compat-wireless releases.

And you are running a 3.4.x kernel on the `amazon-thor`. Therefore it should be possible to use the `ath6k` driver, instead of the `compat-wireless` one, right?

Here they describe how to enable that driver (and it's disabled in your current kernel config):

https://wireless.wiki.kernel.org/en/users/drivers/ath6kl/setup

**Comment by steamp0rt** [comment]
@ollieparanoid No. According to hashcode, the compat-wireless is actually needed for this weird chip.

https://github.com/LineageOS/android_device_amazon_hdx-common/blob/cm-14.1/compat-wireless/Android.mk

**Comment by ollieparanoid** [comment]
Okay. Regarding your question on how to build and install the modules, from a quick look at the aports, this seems to be like the other devices are doing it. Does it not work (do you not have the module file inside the package in the end? after installing, you can list the files with `apk info -L package-name`)

Also is there any reason why you need this as module and not built in? (Other devices are also doing this, so there's probably a reason I'm missing here.)
